lib/internal/bootstrap/loaders.js

```diff
@@ -219,14 +238,25 @@
       undefined;
   };
 
+  const codeCache = getBinding('code_cache');
   NativeModule.prototype.compile = function() {
     let source = NativeModule.getSource(this.id);
     source = NativeModule.wrap(source);
 
     this.loading = true;
 
     try {
-      const script = new ContextifyScript(source, this.filename);
+      // code, filename, lineOffset, columnOffset
+      // cachedData, produceCachedData, parsingContext
+      const script = new ContextifyScript(
+        source, this.filename, 0, 0,
+        codeCache[this.id], false, undefined
+      );
+      if (!codeCache[this.id] || script.cachedDataRejected) {
```

Comment:
I don't think this should happen. If it does, something bad must have happened. Probably best to crash here.

Reply:
@hashseed I assume you are talking about the second condition? The first happens when the cache is not built into the binary. Throwing on the second one (`if (codeCache[this.id] && script.cachedDataRejected)`) sounds reasonable to me.